cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java

```diff
@@ -898,7 +898,7 @@ protected Future<Void> maybeDeleteControllerZnode() {
      */
     protected Future<Void> deleteControllerZnode() {
         // migration rollback process ongoing
-        String zkConnectionString = KafkaResources.zookeeperServiceName(reconciliation.name()) + ":" + ZookeeperCluster.CLIENT_TLS_PORT;
+        String zkConnectionString = DnsNameGenerator.serviceDnsNameWithClusterDomain(reconciliation.namespace(), KafkaResources.zookeeperServiceName(reconciliation.name()))  + ":" + ZookeeperCluster.CLIENT_TLS_PORT;
```

Comment:
Do you really want to use the cluster domain? Wouldn't `serviceDnsNameWithoutClusterDomain` be better?

Reply:
That seems to be also what the scaling is using for example (that uses Pod-based address for connecting to a single pod, but without the cluster domain which is the part relevant here)

Reply:
Yeah cluster domain is actually not needed, I will change. But out of curiosity, is there any specific issue you see by having it?

Reply:
Well, the detection of the cluster domain can be tricky. So we try to avoid relying on it.

Reply:
Done